src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -559,7 +568,55 @@ yield return (new FullBackup.StorageEnvironmentInformation()
 
         public void FullBackupTo(string backupPath)
         {
-            BackupMethods.Full.ToFile(GetAllStoragesEnvironmentInformation(), backupPath);
+            using (var file = new FileStream(backupPath, FileMode.Create))
+            using (var package = new ZipArchive(file, ZipArchiveMode.Create, leaveOpen: true))
+            using (_serverStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var database = _serverStore.Cluster.ReadDatabase(context, Name);
+                Debug.Assert(database != null);
+
+                var zipArchiveEntry = package.CreateEntry(RestoreSettings.FileName, CompressionLevel.Optimal);
+                using (var zipStream = zipArchiveEntry.Open())
+                using (var writer = new BlittableJsonTextWriter(context, zipStream))
+                {
+                    writer.WriteStartObject();
+
+                    // save the database record
+                    writer.WritePropertyName(nameof(RestoreSettings.DatabaseRecord));
+                    var databaseRecordBlittable = EntityToBlittable.ConvertEntityToBlittable(database, DocumentConventions.Default, context);
+                    context.Write(writer, databaseRecordBlittable);
+
+                    // save the database values (subscriptions, periodic backups statuses, etl states...)
+                    writer.WriteComma();
+                    writer.WritePropertyName(nameof(RestoreSettings.DatabaseValues));
+                    writer.WriteStartObject();
+
+                    var first = true;
+                    foreach (var keyValue in ClusterStateMachine.ReadValuesStartingWith(context, 
```

Comment:
If the database is encrypted, I'll use the MasterKey.
Should this be encrypted anyway?